qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1342,6 +1342,13 @@ def replace_block_with_op(
         block_cargs.sort(key=wire_pos_map.get)
         new_node = DAGOpNode(op, block_qargs, block_cargs, dag=self)
 
+        # check the op to insert matches the number of qubits we put it on
+        if op.num_qubits != len(block_qargs):
+            raise DAGCircuitError(
+                f"Number of qubits in the replacement operation ({op.num_qubits}) does not match "
+                f"the number of qubits ({len(block_qargs)})!"
```

Comment:
This is a bit more explicit:
```suggestion
                f"Number of qubits in the replacement operation ({op.num_qubits}) is not equal to "
                f"the number of qubits in the block ({len(block_qargs)})!"
```